### ISSUE TITLE
Don't make list pages a list of sections. Items should just have a section property

### ISSUE THREAD

**Issue by zadjii-msft** [zadjii-msft/PowerToys#99]
This is me reading between the lines from Teams chats I missed.

As currently written (10/21), IListPage.GetItems returns a list of ISections, each Section filled with items. 

That, as it turns out, is really dumb and hard to consume. So let's just not. 

It's easier to stick another property on IListItem that represents the grouping. Then we can very easily map that property to groups.

/cc @michael-hawker

**Comment by zadjii-msft** [comment]
> Kristen and I were also discussion if extensions just return items and items have a group string or something and the UI does the grouping vs. extensions having to structure their data. Then it could just be `public async Task<IAsyncEnumberable<IListItem>> Getitems()` or something (maybe we need an IAsyncOperation for WinRT), but then the extension author could `yield return` here maybe as results come in and get transposed to the item from the other data model?
 
(And `IListItem` would have a `Group` string or identifier) 

Doc ref: [C# - Iterating with Async Enumerables in C# 8 | Microsoft Learn](https://learn.microsoft.com/en-us/archive/msdn-magazine/2019/november/csharp-iterating-with-async-enumerables-in-csharp-8)

**Comment by zadjii-msft** [comment]
Other docs:
* [How to group items in a list or grid (XAML)
](https://learn.microsoft.com/en-us/previous-versions/windows/apps/hh780627(v=win.10))
* https://learn.microsoft.com/en-us/uwp/api/windows.ui.xaml.controls.groupstyle?view=winrt-26100

**Comment by joadoumie** [comment]
Pinging this thread -- I was going to create a separate issue for reintroducing sections into the API and host UI, but stumbled upon this one which may just be the right one. 

TLDR -- we used to have sections, I liked them, they are gone now. They should be optional for sure, but we should readd this imo. I can create a separate issue if we want.

**Comment by michael-hawker** [comment]
Yeah, originally there were a lot of moving parts. And Filtering + Grouping + Sorting in the UI is a hard problem with WinUI, so we wanted to sort through how the data was going to flow and work with like incremental loading, etc...

I think most of that is sorted out now, so the main thing would be how to add grouping back on. I think we basically just have the 'group' property on items, so in the UI layer, we need to separate out those things and have the UI handle that (and still play nice with filtering).

It could be trying to go back to having ListView do everything and using ObservableGroups and CollectionViews or the Advanced one in the Toolkit or something.

OR

It could be that we have multiple collections and composite the sections within an outer itemscontrol/repeater with individual group headers and lists for each section, and then add the navigation between them at the bottom/top... This may be made easier if we chunk the objects and use data/style selectors so that it's all still technically part of nested items repeaters, as I think they can handle some navigation stuff like that more easily...

It's definitely doable in a couple of ways, but maybe something that needs some munging. It's been a big thorn of just something that needs a good sample on too (or to just get fixed in CollectionViewSource in the platform).
